src/flows/pipes/QueryBuilder/context.tsx

```diff
@@ -268,20 +266,35 @@ export const QueryBuilderProvider: FC = ({children}) => {
       ? EXTENDED_TAG_LIMIT
       : DEFAULT_TAG_LIMIT
 
-    // TODO: Use the `v1.tagKeys` function from the Flux standard library once
-    // this issue is resolved: https://github.com/influxdata/flux/issues/1071
-    query(
-      `${_source}
-              |> range(${formatTimeRangeArguments(range)})
-              |> filter(fn: (r) => ${tagString})
-              |> keys()
-              |> keep(columns: ["_value"])
-              |> distinct()${searchString}${previousTagString}
-              |> filter(fn: (r) => r._value != "_time" and r._value != "_start" and r._value !=  "_stop" and r._value != "_value")
-              |> sort()
-              |> limit(n: ${limit})`,
-      scope
-    )
+    let queryText = `${_source}
+    |> range(${formatTimeRangeArguments(range)})
+    |> filter(fn: (r) => ${tagString})
+    |> keys()
+    |> keep(columns: ["_value"])
+    |> distinct()${searchString}${previousTagString}
+    |> filter(fn: (r) => r._value != "_time" and r._value != "_start" and r._value !=  "_stop" and r._value != "_value")
+    |> sort()
+    |> limit(n: ${limit})`
+
+    if (
+      data.buckets[0].type !== 'sample' &&
+      isFlagEnabled('MeasurementFieldsNoTime')
+    ) {
+      _source = `import "regexp"
+      import "influxdata/influxdb/schema"`
+      queryText = `${_source}
+  schema.tagKeys(
+    bucket: "${data.buckets[0].name}",
+    predicate: (r) => ${tagString},
+    start: ${CACHING_REQUIRED_START_DATE},
+    stop: ${CACHING_REQUIRED_END_DATE},
+  )
+    |> distinct()${searchString}${previousTagString}
+    |> filter(fn: (r) => r._value != "_time" and r._value != "_start" and r._value !=  "_stop" and r._value != "_value")
+    |> sort()`
```

Comment:
I see that there was a `limit` appended to the old query that's omitted here, you need it to get the same answer when the feature flag is toggled on/off, yes?

Reply:
Curious why you prefer `limit |> sort` over `sort |> limit`?

The former means that you are truncating a result to the first `n` rows, and then sorting them. This result might not be deterministic; it could be different every time. I'm actually not sure what order storage will choose to return tag values and tag keys. It is better for performance though, since you are sorting fewer things in Flux.

Doing `sort |> limit` will be deterministic, but if a lot of tag keys or values are being returned it could take longer. I don't think it would make that much of a difference though.

My sense is that users would prefer to have a deterministic result. Just my $.02.

Reply:
> I'm actually not sure what order storage will choose to return tag values and tag keys.

should be lexically sorted - the only way we can merge results from multiple shards. 

In any case, sort -> limit is usually the pattern for determinism as Chris mentions.

Reply:
yup, I was done for the day. So, was going to change that today: https://github.com/influxdata/ui/pull/4426#issuecomment-1103011806

Reply:
If it's already sorted, then should I remove sort all together from these new queries?

Reply:
wait a sec, i think the limit |> sort was done for a reason. a dumb reason. but a reason

Reply:
if there is a huge data set, doesnt sort |> limit cause the query to iterate over the entire dataset before limiting the result? wouldn't switching to sort |> limit cause additional timeouts for our users? it's something we're constantly sensitive about. we should def test @philjb's assumption as just adding a limit would be the best of all worlds, but i dont think that query was constructed like that for deterministic results but to save the sorting from happening on the frontend

Reply:
@philjb

Reply:
@drdelambre As Phil says, the sorting and limiting happens in the Flux engine, and in addition we have an optimization where if we see a `sort |> limit` we will do a special sort that only needs to track the top _n_ records. 

So while it is still expensive to sort many things, it may be faster than you think, especially if _n_ is something small. In my experiments, UI metaqueries seem to do `limit(n: 200)` which is quite reasonable; I wouldn't be that concerned about performance in this case.